kubernetes/istio_details_service.go

```diff
@@ -439,8 +440,16 @@ func FilterByRoute(spec map[string]interface{}, protocols []string, service stri
 										if destinationW, ok := mDestination["destination"]; ok {
 											if mDestinationW, ok := destinationW.(map[string]interface{}); ok {
 												if host, ok := mDestinationW["host"]; ok {
-													if sHost, ok := host.(string); ok && FilterByHost(sHost, service, namespace) {
-														return true
+													if sHost, ok := host.(string); ok {
+														if FilterByHost(sHost, service, namespace) {
+															return true
+														}
+														if serviceEntries != nil {
+															// We have ServiceEntry to check
+															if _, found := serviceEntries[strings.ToLower(protocol)+sHost]; found {
```

Comment:
I guess I see a problem here.
This map is protocol+host, where protocol from the map is one of these [1]:
HTTP|HTTPS|GRPC|HTTP2|MONGO|TCP|TLS

But the protocol in the VirtualService will be one of [2]:
http, tls or tcp

Then something is not going to work when we test other usecases beyond http - I think -.

Perhaps in the ServiceEntryHostnames() method it should map the protocols used into the three ones expected by the VirtualServices ?

[1] https://istio.io/docs/reference/config/istio.networking.v1alpha3/#Port
[2] https://istio.io/docs/reference/config/istio.networking.v1alpha3/#VirtualService

Reply:
Our NoHostChecker does not actually understand "TLS" at all at this point, only "http/tcp":

https://github.com/kiali/kiali/blob/master/services/business/checkers/virtual_services/no_host_checker.go#L20

So should I change that too?

Reply:
TLS is being addressed by @xeviknal in a different PR (I guess there is a PR with same work).